
Signed-off-by: Eugene Scherbinin <escherbinin@yandex.ru>
---
Index: target/linux/ramips/image/Makefile
===================================================================
--- target/linux/ramips/image/Makefile	(revision 35342)
+++ target/linux/ramips/image/Makefile	(working copy)
@@ -224,6 +224,17 @@
         $(call BuildFirmware/Generic/initramfs,$(1),$(2),$(call mkcmdline,$(3),$(4),$(5)) $(call mkmtd/$(6),$(mtdlayout_dir615h1)))
 endef
 
+mtdlayout_dir320b1=192k(u-boot)ro,64k(u-boot-env)ro,64k(factory)ro,896k(kernel),6976k(rootfs),7872k@0x50000(firmware)
+mtd_dir320b1_kernel_part_size=917504
+mtd_dir320b1_rootfs_part_size=7143424
+define BuildFirmware/DIR320B1
+	$(call BuildFirmware/Generic,$(1),$(2),$(call mkcmdline,$(3),ttyS1,57600) $(call mkmtd/$(6),$(mtdlayout_dir320b1)),$(mtd_dir320b1_kernel_part_size),$(mtd_dir320b1_rootfs_part_size))
+endef
+
+define BuildFirmware/DIR320B1/initramfs
+	$(call BuildFirmware/Generic/initramfs,$(1),$(2),$(call mkcmdline,$(3),ttyS1,57600) $(call mkmtd/phys,$(mtdlayout_dir320b1)),$(mtd_dir320b1_kernel_part_size),$(mtd_dir320b1_rootfs_part_size))
+endef
+
 mtdlayout_dap1350=192k(u-boot)ro,64k(devdata)ro,192k(devconf)ro,256k(devlang)ro,1088k(kernel),6400k(rootfs),7488k@0xb0000(firmware)
 mtd_dap1350_kernel_part_size=1114112
 mtd_dap1350_rootfs_part_size=6553600
@@ -512,6 +523,10 @@
         $(call Image/Build/Template/$(fs_squash)/$(1),DIR615H1,dir-615-h1,DIR-615-H1,ttyS1,57600,spi)
 endef
 
+define Image/Build/Profile/DIR320B1
+        $(call Image/Build/Template/$(fs_squash)/$(1),DIR320B1,dir-320-b1,DIR-320-B1,ttyS1,57600,spi)
+endef
+
 define Image/Build/Profile/DAP1350
 	$(call Image/Build/Template/$(fs_squash)/$(1),DAP1350,dap-1350,DAP-1350,RT3052-AP-DAP1350-3)
 endef
@@ -649,6 +664,7 @@
 	$(call Image/Build/Profile/CARAMBOLA,$(1))
 	$(call Image/Build/Profile/DIR300B1,$(1))
 	$(call Image/Build/Profile/DIR615H1,$(1))
+	$(call Image/Build/Profile/DIR320B1,$(1))
 	$(call Image/Build/Profile/DAP1350,$(1))
 	$(call Image/Build/Profile/ESR9753,$(1))
 	$(call Image/Build/Profile/F5D8235V2,$(1))
